subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -262,6 +262,19 @@ static int pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 	struct osdp_cmd cmd;
 	struct osdp_event *event;
 
+	if (pd->resend) {
+		pd->resend = false;
+		if (pd->cmd_id != pd->last_cmd_meta.cmd_id) {
+			pd->ephemeral_data[0] = OSDP_PD_NAK_CMD_UNKNOWN;
```

Comment:
A more appropriate NAK code would be `OSDP_PD_NAK_SEQ_NUM`. Since, the command is not the same as the last one but the sequence number hasn't moved.